src/java/org/apache/cassandra/cql3/UpdateParameters.java

```diff
@@ -163,8 +163,28 @@ public Cell<?> addCell(ColumnMetadata column, ByteBuffer value) throws InvalidRe
 
     public Cell<?> addCell(ColumnMetadata column, CellPath path, ByteBuffer value) throws InvalidRequestException
     {
+        // General column value size
         Guardrails.columnValueSize.guard(value.remaining(), column.name.toString(), false, clientState);
 
+        if (column.type.asCQL3Type() == CQL3Type.Native.ASCII) // Ascii size specific guardrail
```

Comment:
Why not a switch statement here? if not, no need for the else's with the nested ifs, it makes it more verbose -- use else if instead. Also think moving this to a `validateSize` function (or something similarly named) would make it more clear